gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -124,6 +124,32 @@ public synchronized int read(@Nonnull byte[] buf, int offset, int length) throws
     return response;
   }
 
+  @Override
+  public void readFully(long position, byte[] buffer, int offset, int length) throws IOException {
+    streamStatistics.readFullyOperationStarted(position, length);
+    checkNotNull(buffer, "buf must not be null");
```

Comment:
consider adding additional details like position, offset and length

Reply:
Also this check is not done by validatePositionedReadArgs?

Reply:
validatePositionedRead doesn't check for null buffer. Rest of checks are done by validatePositionedReadArgs.